src/transport/WsOutboundTransporter.ts

```diff
@@ -0,0 +1,98 @@
+import { OutboundTransporter } from './OutboundTransporter'
+import { Agent } from '../agent/Agent'
+import { WebSocketTransport } from '../agent/TransportService'
+import { Logger } from '../logger'
+import { ConnectionRecord } from '../modules/connections'
+import { OutboundPackage } from '../types'
+import { Symbols } from '../symbols'
+import { WebSocket } from '../utils/ws'
+
+export class WsOutboundTransporter implements OutboundTransporter {
```

Comment:
This is only the WebSocket outbound transporter from the client-side perspective. It could be good to mention it somewhere. Folder structure `src/transport/client`, name of the class `WsClientOutboundTransporter`? Or `edge`/`holder` I'm not sure 🤷‍♂️  😄

Reply:
Maybe I'm not understanding correctly.

But there is not necessarily a difference between a client / edge / holder outbound transport and another outbound transport. That should be the concern of the higher level. Outbound transport should just send (and optionally receive) messages to an endpoint. Same with HTTP outbound transport.

Reply:
To clarify further. On the server side it would be the `WsInboundTransporter` that hosts the WebSocket

Reply:
The naming of inbound and outbound is confusing, but is from the standpoint of transport being simplex by default in Aries

Reply:
So outbound transport is you initiate the interaction (so for HTTP and WS that means you're the client, server can't initiate the interaction)

For inbound transport it means the other agent initiates the interaction (so for HTTP and WS that means you host the server/endpoint, you can't initiate the interaction)

Reply:
Ah, sorry. I realized I was too focused on the concrete implementation for edge agent and mediation where the logic differs, but the one doesn't exclude the other 👍